include/mxnet/storage.h

```diff
@@ -78,6 +78,16 @@ class Storage {
    */
   virtual ~Storage() {}
   /*!
+   * \brief Returns mutex used by storage manager
+   */
+  std::mutex& GetMutex(Context::DeviceType dev) {
```

Comment:
Could you add brief description when mutex is required?

Reply:
Mutex is not really required outside of NCCL (and only for GPU allocations). See discussion from the previous NCCL PR: https://github.com/apache/incubator-mxnet/pull/5521#issuecomment-288210480